Modules/Core/Common/test/itkSobelOperatorImageConvolutionTest.cxx

```diff
@@ -65,19 +68,98 @@ itkSobelOperatorImageConvolutionTest(int argc, char * argv[])
+  typename ImageType::PixelType center_value = output3x3Image->GetPixel(center_index);
+  if (center_value != 0)
+  {
+    std::cout << "ERROR: Constant image convolution with SobelOperator should return 0, "
+              << "but value of " << +center_value << " was computed. [" << pixelType << "]" << std::endl;
+    return EXIT_FAILURE;
+  }
+  return EXIT_SUCCESS;
+}
+
+int
+itkSobelOperatorImageConvolutionTest(int argc, char * argv[])
+{
+  if (argc != 4)
+  {
+    std::cerr << "Missing parameters." << std::endl;
+    std::cerr << "Usage: " << itkNameOfTestExecutableMacro(argv) << " inputFileName direction outputFileName"
+              << std::endl;
+    return EXIT_FAILURE;
+  }
+  const auto direction = std::stoul(argv[2]);
+  int        return_status = EXIT_SUCCESS;
+
+  constexpr unsigned int Dimension = 2;
+  /* Sanity Checking For Sobel Operator */
+  //{
+  /*
+   * Demonstrate that signed types do **NOT** work with SobelOperator
+   * The unsigned checks require disableing Signed ConceptChecking in SobelOperator
+   * added in March 2023.
+   */
+  /*
+  return_status += DoSimpleConvolutionTest<unsigned char, 2>(direction, "unsigned char");
+  return_status += DoSimpleConvolutionTest<unsigned short, 2>(direction, "unsigned short");
+  return_status += DoSimpleConvolutionTest<unsigned int, 2>(direction, "unsigned int");
+  return_status += DoSimpleConvolutionTest<unsigned long, 2>(direction, "unsigned long");
+  */
+  //}
+  {
+    /*
+     * Demonstrate that signed types do work with SobelOperator
+     */
+    return_status += DoSimpleConvolutionTest<char, 2>(direction, "char");
+    return_status += DoSimpleConvolutionTest<short, 2>(direction, "short");
+    return_status += DoSimpleConvolutionTest<int, 2>(direction, "int");
+    return_status += DoSimpleConvolutionTest<long, 2>(direction, "long");
+    return_status += DoSimpleConvolutionTest<float, 2>(direction, "float");
+    return_status += DoSimpleConvolutionTest<double, 2>(direction, "double");
+  }
+
+  {
+    /* Test on a real image */
+    using PixelType = int16_t;
+    using ImageType = itk::Image<PixelType, Dimension>;
+
+    const auto inputImage = itk::ReadImage<ImageType>(argv[1]);
+
+    auto signedSobelImage = DoConvolution<ImageType>(inputImage, direction);
+
+    using OutputImageType = itk::Image<uint8_t, Dimension>;
+    // Assume min/max values are approximately +/- same magnitude so that the output images
+    // to be stored in uint8_t have an implied 0 at about pixel value 128.  Many web based viewers
+    // for the difference images in the testing outputs render better in this positive png range.
+    using RescaleIntensityType = itk::RescaleIntensityImageFilter<ImageType, OutputImageType>;
+    RescaleIntensityType::Pointer rescalerForVisualization = RescaleIntensityType::New();
+    rescalerForVisualization->SetInput(signedSobelImage);
+    rescalerForVisualization->SetOutputMinimum(0);
+    rescalerForVisualization->SetOutputMaximum(255);
```

Comment:
Why change test output type from png to tiff, if we rescale to 0-255 range?

Reply:
Originally I was writing signed sobel outputs, but only Slicer would display those outputs with reasonable visualization.  I wanted the outputs to display in cdash as a better looking (i.e. look like sobel should look), so at the last minute I updated to rescale the images.  

Since the images needed to be made new, I did not think that tiff would be any worse than png.

Reply:
When I first saw TIFF, and commit message mentioning negative gradients, I thought negative values were going to be in the test images 😄

A bit surprising, that's all.